### PR TITLE
Add Via support for TG4x

### PR DESCRIPTION
**Pull request by redsn0w422**
## Description

- Adds a new keymap with Via enabled for the TG4x.
- Updates the default keymap to have a mapping for `RESET`.
- Updates the Vendor ID.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by gwillad** [comment]
someone in a discord wanted to try to use via for their tg4x. Can confirm it works correctly

**Review by drashna — APPROVED** [review]

File: keyboards/tg4x/keymaps/via/config.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2019 MechMerlin
```
Comment (fauxpark) [inline keyboards/tg4x/keymaps/via/config.h:1]: This file can be deleted.

File: keyboards/tg4x/keymaps/default/keymap.c [hunk 2]
```diff
@@ -20,7 +20,28 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [3] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
     ),
```
Comment (fauxpark) [inline keyboards/tg4x/keymaps/default/keymap.c:45]:
```suggestion
    )
```

File: keyboards/tg4x/keymaps/via/rules.mk [hunk 3]
```diff
@@ -0,0 +1,4 @@
+RGBLIGHT_ENABLE = yes 		# LED strips
+VIA_ENABLE = yes
+LTO_ENABLE = yes
+MOUSEKEY_ENABLE = no
```
Comment (fauxpark) [inline keyboards/tg4x/keymaps/via/rules.mk:4]:
```suggestion
```

File: keyboards/tg4x/config.h [hunk 4]
```diff
@@ -20,7 +20,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "config_common.h"
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID 0xFEED
+#define VENDOR_ID 0x7351 // initially added in PR #7351
 #define PRODUCT_ID 0x0001
```
Comment (fauxpark) [inline keyboards/tg4x/config.h:24]:
```suggestion
#define VENDOR_ID 0x4D4D // "MM"
#define PRODUCT_ID 0x0458 // 4 "X"
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/tg4x/keymaps/default/keymap.c (hunk 2 above)
Comment (redsn0w422) [reply keyboards/tg4x/keymaps/default/keymap.c:45]: @fauxpark I thought it was recommended to have the 4 layers pre-created for via firmwares?

Comment (fauxpark) [reply keyboards/tg4x/keymaps/default/keymap.c:45]: This isn't the VIA keymap.

Comment (redsn0w422) [reply keyboards/tg4x/keymaps/default/keymap.c:45]: Ah, good point :)

**Review by fauxpark — APPROVED** [review]

**Pull request merged**